packages/instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -272,47 +291,105 @@ export class AwsLambdaInstrumentation extends InstrumentationBase<AwsLambdaInstr
       };
     }
 
-    return function patchedHandler(
-      this: never,
-      // The event can be a user type, it truly is any.
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      event: any,
-      context: Context,
-      callback: Callback
-    ) {
-      _onRequest();
-
-      const parent = plugin._determineParent(event, context);
-
-      const span = plugin._createSpanForRequest(
-        event,
-        context,
-        requestIsColdStart,
-        parent
-      );
+    // Determine runtime version to decide whether to support callbacks
+    const nodeVersion = getNodeRuntimeVersion();
+    const supportsCallbacks = nodeVersion === null || nodeVersion < 24;
```

Comment:
since we don't use `getNodeRuntimeVersion` elsewhere, what do you think about removing the fact that it can return null and rename it to "isSupportingCallbacks()"
something like
```nodejs
const isSupportingCallbacks = (): boolean {
  const executionEnv = process.env.AWS_EXECUTION_ENV;
  if (!executionEnv) {
    return true;
  }

  // AWS_EXECUTION_ENV format: AWS_Lambda_nodejs24.x, AWS_Lambda_nodejs22.x, etc.
  const match = executionEnv.match(/AWS_Lambda_nodejs(\d+)\./);
  if (match && match[1]) {
    return parseInt(match[1], 10) < 24
  }

  return true;
}
```